modules/oe_theme_content_event/oe_theme_content_event.module

```diff
@@ -0,0 +1,478 @@
+  $current_time = \Drupal::time()->getRequestTime();
+
+  // Prepare the registration button.
+  $registration_button = [
+    'label' => t('Register'),
+    'state' => 'disabled',
+    'url' => $node->get('oe_event_registration_url')->uri,
+  ];
+  $event_status = $event_status = $node->get('oe_event_status')->value;
+
+  // Get registration dates if any registration status is set.
+  if ($registration_status = $node->get('oe_event_registration_status')->value) {
+    /** @var \Drupal\oe_theme\ValueObject\DateValueObject $registration_dates */
+    $registration_dates = DateValueObject::fromArray([
+      'start' => $node->get('oe_event_registration_start_date')->date->getTimestamp(),
+      'end' => $node->get('oe_event_registration_end_date')->date->getTimestamp(),
+    ]);
+  }
+
+  // No registration information are set.
+  if (!$registration_status && $event_status !== 'cancelled') {
+    $registration_button['description'] = t('Registration is not yet opened for this event.');
+  }
+
+  if (isset($registration_dates) &&
+    ($registration_status === 'open' || $registration_status === 'closed') &&
+    $event_status !== 'cancelled' &&
+    $event_status !== 'postponed'
+  ) {
+    // Registration in the future.
+    if ($registration_dates->getStartTime() > $current_time) {
+      $registration_button['description'] = t('Registration will open on @start_date, until @end_date.', [
+        '@start_date' => $date_formatter->format($registration_dates->getStartTime(), 'oe_theme_event_date_hour'),
+        '@end_date' => $date_formatter->format($registration_dates->getEndTime(), 'oe_theme_event_date_hour'),
+      ]);
+    }
+    // Registration is active.
+    if ($registration_dates->getStartTime() < $current_time && $registration_dates->getEndTime() > $current_time) {
+      $registration_button['state'] = 'active';
+      // Difference between two dates in days.
+      $date_diff = $date_formatter->formatTimeDiffUntil($registration_dates->getEndTime(), ['granularity' => 1]);
+      $registration_button['description'] = t('Book your seat, @time_left left to register.', [
+        '@time_left' => $date_diff,
+      ]);
+    }
+    // Registration is closed.
+    elseif ($registration_dates->getStartTime() < $current_time && $registration_dates->getEndTime() < $current_time) {
+      $registration_button['description'] = t('Registration for this event has ended.');
+    }
+  }
+
+  return $registration_button;
+}
+
+/**
+ * Prepares the practical information section as an array for ECL.
+ *
+ * @param \Drupal\node\NodeInterface $node
+ *   The event node object.
+ *
+ * @return array
+ *   The prepared array.
+ */
+function _oe_theme_content_event_prepare_practical_information(NodeInterface $node): array {
+  /** @var \Drupal\Core\Datetime\DateFormatter $date_formatter */
+  $date_formatter = \Drupal::service('date.formatter');
+  /** @var \Drupal\Core\Render\Renderer $renderer */
+  $renderer = \Drupal::service('renderer');
```

Comment:
This doesn't seem like a good practice, since this is a preprocess method, why not use the render arrays?

Reply:
We have to do it here, because ECL twig does not support render arrays. I checked this with @ademarco